src/api/student.js

```diff
@@ -20,6 +20,12 @@ app.get('/all', (req, res) => {
     res.successJson(StudentStore.getStudents());
 });
 
+app.patch('/', (req, res) => {
+    addAction('UPDATE_STUDENT', req.body).then(() => {
+        res.successJson(StudentStore.getStudentById(req.body.id));
```

Comment:
Patch requests usually have the "identifier" in the query not the body of the request. Just change the id to be in the request not the body and it should be fine. Also make sure to change the functional test to not pass a null and the id of the student.